Implement <details>/<summary> disclosure triangle as a list item

nt1m opened the pull request:
#### cd2f382c7dc0e50ae049f7e2420377e944ba60a8
<pre>
Implement &lt;details&gt;/&lt;summary&gt; disclosure triangle as a list item
<a href="https://bugs.webkit.org/show_bug.cgi?id=157323">https://bugs.webkit.org/show_bug.cgi?id=157323</a>
<a href="https://rdar.apple.com/95148788">rdar://95148788</a>

Reviewed by Ryosuke Niwa.

<a href="https://html.spec.whatwg.org/#the-details-and-summary-elements">https://html.spec.whatwg.org/#the-details-and-summary-elements</a>

This does the following changes to match the latest HTML spec:
- Implements details/summary disclosure triangle as a list item
- Implements collapsing of summary with `content-visibility: hidden` as mandated by spec, instead of appending/deleting the default slot
- Sets `display: block` on the default slot when the details element is open for easier styling
- Adds the user-agent styles for details/summary
- Stops forcing the used display value for details/summary
- Removes DetailsMarkerControl/RenderDetailsMarker (::-webkit-details-marker) since this is now covered by the list item ::marker

As a result, the following are fixed:
- ::marker can now be used instead of ::-webkit-details-marker (bug 157323)
- Changing list-style and display is now effective on the ::marker (bug 264381)
- element.checkVisibility() now works properly on elements in the default slot (bug 264733)
- summary::before now appears in the correct order (bug 61426)
- details::before/after now appear in the correct order (bug 61766)
- and more...

* LayoutTests/TestExpectations:
* LayoutTests/accessibility/ios-simulator/aria-details-toggle-summary-expected.txt:
* LayoutTests/fast/css-grid-layout/grid-was-populated-assert-expected.txt:
* LayoutTests/fast/html/details-clone-expected.txt: Removed.
* LayoutTests/fast/html/details-clone.html: Removed.
* LayoutTests/fast/html/details-inline-expected.html: Removed.
* LayoutTests/fast/html/details-inline.html: Removed.
* LayoutTests/fast/multicol/crash-in-vertical-writing-mode-expected.txt:
* LayoutTests/fast/multicol/crash-in-vertical-writing-mode.html:
* LayoutTests/http/tests/security/cross-origin-blob-transfer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/run/inserthorizontalrule-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-blockification-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-display-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-details-element/closed-details-layout-apis.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-details-element/toggleEvent-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-summary-element/anchor-with-inline-element-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-summary-element/interactive-content-expected.txt:
* LayoutTests/platform/glib/fast/css-generated-content/details-summary-before-after-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-details-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-details-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-1-and-click-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-10-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-10-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-2-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-3-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-3-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-4-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-4-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-5-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-5-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-6-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-6-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-7-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-7-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-8-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-8-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-9-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-9-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-marker-style-expected.txt:
* LayoutTests/platform/glib/fast/html/details-marker-style-mixed-expected.txt:
* LayoutTests/platform/glib/fast/html/details-nested-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-nested-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-no-summary1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-no-summary2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-no-summary3-expected.txt:
* LayoutTests/platform/glib/fast/html/details-open1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-open3-expected.txt:
* LayoutTests/platform/glib/fast/html/details-open5-expected.txt:
* LayoutTests/platform/glib/fast/html/details-open6-expected.txt:
* LayoutTests/platform/glib/fast/html/details-position-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-1-and-click-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-2-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-3-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-3-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-4-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-4-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-5-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-5-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-6-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-6-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-writing-mode-expected.txt:
* LayoutTests/platform/glib/fast/html/details-writing-mode-mixed-expected.txt:
* LayoutTests/platform/glib/fast/multicol/span/clone-summary-expected.txt:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/gtk/fast/html/details-add-summary-10-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-2-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-3-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-4-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-5-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-6-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-7-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-8-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-9-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-open2-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-open4-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-remove-summary-2-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-remove-summary-3-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-remove-summary-4-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-remove-summary-5-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-remove-summary-6-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-replace-summary-child-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-replace-text-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/fast/html/details-add-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-details-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-details-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-1-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-10-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-10-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-2-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-3-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-3-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-4-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-4-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-5-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-5-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-6-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-6-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-7-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-7-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-8-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-8-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-9-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-9-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-marker-style-expected.txt:
* LayoutTests/platform/ios/fast/html/details-marker-style-mixed-expected.txt:
* LayoutTests/platform/ios/fast/html/details-nested-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-nested-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-no-summary1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-no-summary2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-no-summary3-expected.txt:
* LayoutTests/platform/ios/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open3-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open4-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open5-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open6-expected.txt:
* LayoutTests/platform/ios/fast/html/details-position-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-1-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-2-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-3-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-3-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-4-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-4-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-5-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-5-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-6-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-6-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-replace-summary-child-expected.txt:
* LayoutTests/platform/ios/fast/html/details-replace-text-expected.txt:
* LayoutTests/platform/ios/fast/html/details-writing-mode-expected.txt:
* LayoutTests/platform/ios/fast/html/details-writing-mode-mixed-expected.txt:
* LayoutTests/platform/ios/fast/multicol/span/clone-summary-expected.txt:
* LayoutTests/platform/mac-ventura/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/mac-ventura/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/mac-ventura/fast/html/details-open2-expected.txt:
* LayoutTests/platform/mac-ventura/fast/html/details-open4-expected.txt:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/mac/fast/css-generated-content/details-summary-before-after-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-details-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-details-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-1-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-10-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-10-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-2-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-3-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-3-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-4-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-4-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-5-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-5-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-6-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-6-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-7-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-7-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-8-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-8-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-9-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-9-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-marker-style-expected.txt:
* LayoutTests/platform/mac/fast/html/details-marker-style-mixed-expected.txt:
* LayoutTests/platform/mac/fast/html/details-nested-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-nested-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-no-summary1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-no-summary2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-no-summary3-expected.txt:
* LayoutTests/platform/mac/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open3-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open4-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open5-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open6-expected.txt:
* LayoutTests/platform/mac/fast/html/details-position-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-1-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-2-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-3-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-3-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-4-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-4-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-5-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-5-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-6-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-6-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-replace-summary-child-expected.txt:
* LayoutTests/platform/mac/fast/html/details-replace-text-expected.txt:
* LayoutTests/platform/mac/fast/html/details-writing-mode-expected.txt:
* LayoutTests/platform/mac/fast/html/details-writing-mode-mixed-expected.txt:
* LayoutTests/platform/mac/fast/multicol/span/clone-summary-expected.txt:
* LayoutTests/platform/wpe/TestExpectations:
* LayoutTests/platform/wpe/fast/html/details-add-summary-10-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-2-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-3-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-4-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-5-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-6-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-7-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-8-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-9-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-open2-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-open4-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-remove-summary-2-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-remove-summary-3-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-remove-summary-4-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-remove-summary-5-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-remove-summary-6-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-replace-summary-child-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-replace-text-expected.txt:
* LayoutTests/tables/table-col-indent-crash-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::determineAccessibilityRole):
* Source/WebCore/css/CSSPseudoSelectors.json:
* Source/WebCore/css/html.css:
(details, summary):
(details &gt; summary:first-of-type):
(details[open] &gt; summary:first-of-type):
(frameset, frame):
(frameset):
(iframe):
(details): Deleted.
(summary): Deleted.
(summary::-webkit-details-marker): Deleted.
* Source/WebCore/css/process-css-pseudo-selectors.py:
* Source/WebCore/html/HTMLDetailsElement.cpp:
(WebCore::HTMLDetailsElement::didAddUserAgentShadowRoot):
(WebCore::HTMLDetailsElement::attributeChanged):
(WebCore::HTMLDetailsElement::createElementRenderer): Deleted.
* Source/WebCore/html/HTMLDetailsElement.h:
* Source/WebCore/html/HTMLSummaryElement.cpp:
(WebCore::HTMLSummaryElement::create):
(): Deleted.
(WebCore::HTMLSummaryElement::createElementRenderer): Deleted.
(WebCore::HTMLSummaryElement::didAddUserAgentShadowRoot): Deleted.
* Source/WebCore/html/HTMLSummaryElement.h:
* Source/WebCore/html/shadow/DetailsMarkerControl.cpp: Removed.
* Source/WebCore/html/shadow/DetailsMarkerControl.h: Removed.
* Source/WebCore/html/shadow/detailsElementShadow.css: Added.
(:host summary):
(:host([open]) summary):
* Source/WebCore/layout/integration/LayoutIntegrationBoxTree.cpp:
* Source/WebCore/rendering/RenderDetailsMarker.cpp: Removed.
* Source/WebCore/rendering/RenderDetailsMarker.h: Removed.
* Source/WebCore/rendering/RenderTreeAsText.cpp:
(WebCore::RenderTreeAsText::writeRenderObject):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):

Canonical link: <a href="https://commits.webkit.org/286869@main">https://commits.webkit.org/286869@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a25e7d1d6602fded4793982c5d4be8420c6bd09f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/77324 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/56359 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/30240 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/81896 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/28606 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/79441 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/65507 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/4656 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/60580 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/18620 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/80391 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/50555 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/66374 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40879 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47957 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/23873 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/26929 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/69070 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/24211 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/83313 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/4704 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/3195 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/68851 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/4860 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/66345 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/68106 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/17013 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/12107 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/10205 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/4651 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/7466 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/4670 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/8105 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/6429 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/html/HTMLDetailsElement.cpp:
```diff
@@ -117,6 +113,14 @@ void HTMLDetailsElement::didAddUserAgentShadowRoot(ShadowRoot& root)
+    m_defaultSlot->setInlineStyleProperty(CSSPropertyDisplay, CSSValueBlock);
+    root.appendChild(*m_defaultSlot);
+
+    static MainThreadNeverDestroyed<const String> stylesheet(StringImpl::createWithoutCopying(detailsElementShadowUserAgentStyleSheet));
```
Maybe we can use `adoptedStyleSheets()` instead of creating a new style element?

nt1m replied:
The spec uses a style element: https://html.spec.whatwg.org/#the-details-and-summary-elements

annevk replied:
It should be the same though. I said as much in the specification review and the main reason the `style` element approach was allowed because you couldn't observe the difference. If you're setting it up programmatically I'm not entirely sure it makes sense to create the additional node.

nt1m replied:
I can try it, but it'll probably end up inconsistent with meter/attachment, although I guess they were created before `adoptedStylesheets`

nt1m replied:
I tried to address this but none of my attempts worked

twilco commented on Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
```diff
@@ -2127,15 +2127,20 @@ AccessibilityRole AccessibilityRenderObject::determineAccessibilityRole()
+    RefPtr node = m_renderer->node();
+    if (m_renderer->isRenderListItem()) {
+        // Active summaries are handled by AccessibilityNodeObject::determineAccessibilityRoleFromNode().
+        RefPtr summary = dynamicDowncast<HTMLSummaryElement>(node);
+        if (!summary || !summary->isActiveSummary())
```
Could we please add a comment here saying something to the effect of "The details / summary disclosure triangle is implemented using RenderListItem, so check to make sure we don't have one of those." Re-word as you see fit.

nt1m replied:
Will do, once I fix all the EWS failures

nt1m replied:
Fixed!

nt1m commented on LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-blockification-expected.txt:
```diff
@@ -3,5 +3,5 @@ bar
 bar
 
-PASS Summary and content should have display:block computed value
+FAIL Summary and content should have display:block computed value assert_equals: expected "block" but got "inline"
```
This test relies on `::details-content` to emulate the old behavior of details/summary. It'll re-progress once I implement the pseudo-element: https://bugs.webkit.org/show_bug.cgi?id=275211 (I already have a PR for it)

nt1m committed:
Implement <details>/<summary> disclosure triangle as a list item
https://bugs.webkit.org/show_bug.cgi?id=157323
rdar://95148788

Reviewed by Ryosuke Niwa.

https://html.spec.whatwg.org/#the-details-and-summary-elements

This does the following changes to match the latest HTML spec:
- Implements details/summary disclosure triangle as a list item
- Implements collapsing of summary with `content-visibility: hidden` as mandated by spec, instead of appending/deleting the default slot
- Sets `display: block` on the default slot when the details element is open for easier styling
- Adds the user-agent styles for details/summary
- Stops forcing the used display value for details/summary
- Removes DetailsMarkerControl/RenderDetailsMarker (::-webkit-details-marker) since this is now covered by the list item ::marker

As a result, the following are fixed:
- ::marker can now be used instead of ::-webkit-details-marker (bug 157323)
- Changing list-style and display is now effective on the ::marker (bug 264381)
- element.checkVisibility() now works properly on elements in the default slot (bug 264733)
- summary::before now appears in the correct order (bug 61426)
- details::before/after now appear in the correct order (bug 61766)
- and more...

* LayoutTests/TestExpectations:
* LayoutTests/accessibility/ios-simulator/aria-details-toggle-summary-expected.txt:
* LayoutTests/fast/css-grid-layout/grid-was-populated-assert-expected.txt:
* LayoutTests/fast/html/details-clone-expected.txt: Removed.
* LayoutTests/fast/html/details-clone.html: Removed.
* LayoutTests/fast/html/details-inline-expected.html: Removed.
* LayoutTests/fast/html/details-inline.html: Removed.
* LayoutTests/fast/multicol/crash-in-vertical-writing-mode-expected.txt:
* LayoutTests/fast/multicol/crash-in-vertical-writing-mode.html:
* LayoutTests/http/tests/security/cross-origin-blob-transfer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/editing/run/inserthorizontalrule-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-blockification-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/rendering/the-details-element/details-display-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-details-element/closed-details-layout-apis.tentative-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-details-element/toggleEvent-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-summary-element/anchor-with-inline-element-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-summary-element/interactive-content-expected.txt:
* LayoutTests/platform/glib/fast/css-generated-content/details-summary-before-after-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-details-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-details-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-1-and-click-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-10-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-10-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-2-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-3-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-3-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-4-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-4-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-5-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-5-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-6-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-6-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-7-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-7-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-8-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-8-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-9-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-add-summary-9-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-add-summary-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-marker-style-expected.txt:
* LayoutTests/platform/glib/fast/html/details-marker-style-mixed-expected.txt:
* LayoutTests/platform/glib/fast/html/details-nested-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-nested-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-no-summary1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-no-summary2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-no-summary3-expected.txt:
* LayoutTests/platform/glib/fast/html/details-open1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-open3-expected.txt:
* LayoutTests/platform/glib/fast/html/details-open5-expected.txt:
* LayoutTests/platform/glib/fast/html/details-open6-expected.txt:
* LayoutTests/platform/glib/fast/html/details-position-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-1-and-click-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-2-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-3-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-3-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-4-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-4-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-5-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-5-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-6-and-click-expected.txt: Removed.
* LayoutTests/platform/glib/fast/html/details-remove-summary-6-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-child-1-expected.txt:
* LayoutTests/platform/glib/fast/html/details-remove-summary-child-2-expected.txt:
* LayoutTests/platform/glib/fast/html/details-writing-mode-expected.txt:
* LayoutTests/platform/glib/fast/html/details-writing-mode-mixed-expected.txt:
* LayoutTests/platform/glib/fast/multicol/span/clone-summary-expected.txt:
* LayoutTests/platform/gtk/TestExpectations:
* LayoutTests/platform/gtk/fast/html/details-add-summary-10-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-2-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-3-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-4-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-5-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-6-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-7-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-8-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-add-summary-9-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-open2-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-open4-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-remove-summary-2-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-remove-summary-3-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-remove-summary-4-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-remove-summary-5-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-remove-summary-6-and-click-expected.txt: Added.
* LayoutTests/platform/gtk/fast/html/details-replace-summary-child-expected.txt:
* LayoutTests/platform/gtk/fast/html/details-replace-text-expected.txt:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/ios/fast/html/details-add-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-details-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-details-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-1-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-10-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-10-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-2-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-3-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-3-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-4-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-4-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-5-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-5-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-6-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-6-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-7-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-7-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-8-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-8-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-9-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-9-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-add-summary-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-marker-style-expected.txt:
* LayoutTests/platform/ios/fast/html/details-marker-style-mixed-expected.txt:
* LayoutTests/platform/ios/fast/html/details-nested-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-nested-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-no-summary1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-no-summary2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-no-summary3-expected.txt:
* LayoutTests/platform/ios/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open3-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open4-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open5-expected.txt:
* LayoutTests/platform/ios/fast/html/details-open6-expected.txt:
* LayoutTests/platform/ios/fast/html/details-position-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-1-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-2-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-3-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-3-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-4-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-4-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-5-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-5-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-6-and-click-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-6-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-child-1-expected.txt:
* LayoutTests/platform/ios/fast/html/details-remove-summary-child-2-expected.txt:
* LayoutTests/platform/ios/fast/html/details-replace-summary-child-expected.txt:
* LayoutTests/platform/ios/fast/html/details-replace-text-expected.txt:
* LayoutTests/platform/ios/fast/html/details-writing-mode-expected.txt:
* LayoutTests/platform/ios/fast/html/details-writing-mode-mixed-expected.txt:
* LayoutTests/platform/ios/fast/multicol/span/clone-summary-expected.txt:
* LayoutTests/platform/mac-ventura/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/mac-ventura/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/mac-ventura/fast/html/details-open2-expected.txt:
* LayoutTests/platform/mac-ventura/fast/html/details-open4-expected.txt:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/mac/fast/css-generated-content/details-summary-before-after-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-details-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-details-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-1-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-10-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-10-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-2-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-3-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-3-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-4-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-4-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-5-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-5-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-6-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-6-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-7-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-7-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-8-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-8-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-9-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-9-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-add-summary-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-marker-style-expected.txt:
* LayoutTests/platform/mac/fast/html/details-marker-style-mixed-expected.txt:
* LayoutTests/platform/mac/fast/html/details-nested-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-nested-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-no-summary1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-no-summary2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-no-summary3-expected.txt:
* LayoutTests/platform/mac/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open3-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open4-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open5-expected.txt:
* LayoutTests/platform/mac/fast/html/details-open6-expected.txt:
* LayoutTests/platform/mac/fast/html/details-position-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-1-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-2-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-3-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-3-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-4-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-4-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-5-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-5-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-6-and-click-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-6-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-child-1-expected.txt:
* LayoutTests/platform/mac/fast/html/details-remove-summary-child-2-expected.txt:
* LayoutTests/platform/mac/fast/html/details-replace-summary-child-expected.txt:
* LayoutTests/platform/mac/fast/html/details-replace-text-expected.txt:
* LayoutTests/platform/mac/fast/html/details-writing-mode-expected.txt:
* LayoutTests/platform/mac/fast/html/details-writing-mode-mixed-expected.txt:
* LayoutTests/platform/mac/fast/multicol/span/clone-summary-expected.txt:
* LayoutTests/platform/wpe/TestExpectations:
* LayoutTests/platform/wpe/fast/html/details-add-summary-10-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-2-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-3-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-4-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-5-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-6-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-7-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-8-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-add-summary-9-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-no-summary4-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-open-javascript-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-open2-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-open4-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-remove-summary-2-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-remove-summary-3-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-remove-summary-4-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-remove-summary-5-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-remove-summary-6-and-click-expected.txt: Added.
* LayoutTests/platform/wpe/fast/html/details-replace-summary-child-expected.txt:
* LayoutTests/platform/wpe/fast/html/details-replace-text-expected.txt:
* LayoutTests/tables/table-col-indent-crash-expected.txt:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/DerivedSources-input.xcfilelist:
* Source/WebCore/DerivedSources.make:
* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/accessibility/AccessibilityRenderObject.cpp:
(WebCore::AccessibilityRenderObject::determineAccessibilityRole):
* Source/WebCore/css/CSSPseudoSelectors.json:
* Source/WebCore/css/html.css:
(details, summary):
(details > summary:first-of-type):
(details[open] > summary:first-of-type):
(frameset, frame):
(frameset):
(iframe):
(details): Deleted.
(summary): Deleted.
(summary::-webkit-details-marker): Deleted.
* Source/WebCore/css/process-css-pseudo-selectors.py:
* Source/WebCore/html/HTMLDetailsElement.cpp:
(WebCore::HTMLDetailsElement::didAddUserAgentShadowRoot):
(WebCore::HTMLDetailsElement::attributeChanged):
(WebCore::HTMLDetailsElement::createElementRenderer): Deleted.
* Source/WebCore/html/HTMLDetailsElement.h:
* Source/WebCore/html/HTMLSummaryElement.cpp:
(WebCore::HTMLSummaryElement::create):
(): Deleted.
(WebCore::HTMLSummaryElement::createElementRenderer): Deleted.
(WebCore::HTMLSummaryElement::didAddUserAgentShadowRoot): Deleted.
* Source/WebCore/html/HTMLSummaryElement.h:
* Source/WebCore/html/shadow/DetailsMarkerControl.cpp: Removed.
* Source/WebCore/html/shadow/DetailsMarkerControl.h: Removed.
* Source/WebCore/html/shadow/detailsElementShadow.css: Added.
(:host summary):
(:host([open]) summary):
* Source/WebCore/layout/integration/LayoutIntegrationBoxTree.cpp:
* Source/WebCore/rendering/RenderDetailsMarker.cpp: Removed.
* Source/WebCore/rendering/RenderDetailsMarker.h: Removed.
* Source/WebCore/rendering/RenderTreeAsText.cpp:
(WebCore::RenderTreeAsText::writeRenderObject):
* Source/WebCore/style/UserAgentStyle.cpp:
(WebCore::Style::UserAgentStyle::ensureDefaultStyleSheetsForElement):

Canonical link: https://commits.webkit.org/286869@main
